File: solutions/javascript/collatz-conjecture/1/collatz-conjecture.js
export function steps(number) {
  if (number<1){throw new Error('Only positive integers are allowed')}
  if(number===1){
    return 0
  }
  for(let i = 0; i < 200; i++) {
    if (number%2===0){
      number/=2
    }
    else{
      number = (number*3)+1
    }
    if(number===1){
      return i+1
    }
  }
}

File: solutions/go/collatz-conjecture/1/collatz_conjecture.go
package collatzconjecture

import "errors"
func CollatzConjecture(n int) (int, error) {
    if n < 1 {
        return 0, errors.New("number must be positive")
    }
	if n == 1 {
        return 0, nil
    } else if n % 2 == 0 {
      nextStep, _ := CollatzConjecture(n / 2)
      return 1 + nextStep, nil
    }
    nextStep, _ := CollatzConjecture(n * 3 + 1)
    return 1 + nextStep, nil
}
